lua/lazy/manage/rocks.lua

```diff
@@ -0,0 +1,75 @@
+local Config = require("lazy.core.config")
+local Util = require("lazy.core.util")
+
+---@class LazyRock
+---@field plugin string
+---@field name string
+---@field spec string
+---@field installed boolean
+
+local M = {}
+---@type LazyRock[]
+M.rocks = {}
+
+---@param ... string
+---@return string[]
+function M.args(...)
+  local ret = { "--tree", Config.rocks.tree, "--lua-version", "5.1" }
```

Comment:
note: luarocks has a `root `manifest (for packages with semver versions) and a `dev` manifest (for `scm` rockspecs).

I don't know if you want to support installing `dev` (or `scm`) rocks, but if you do, luarocks requires a `--dev` flag to enable searching the `dev` manifest.
The big downside to this is that with the `--dev` flag enabled, it will prioritise `scm` versions for all dependencies that don't have a `<` constraint. Sadly, some rocks only have `scm` versions.

In rocks.nvim, we add the `--dev` flag if the `version` of a rock is `dev` or `scm`.

Reply:
Makes sense. Added!